src/compiler/emitter.ts

```diff
@@ -214,6 +371,26 @@ var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, Promi
                 exportFunctionForFile = undefined;
                 emit(sourceFile);
             }
+            
+            function emitBundledFile(sourceFile: SourceFile): void {
+                let canonicalName = removeFileExtension(
+                    getRelativePathToDirectoryOrUrl(
+                        host.getCurrentDirectory(),
+                        sourceFile.fileName,
+                        host.getCurrentDirectory(),
+                        host.getCanonicalFileName,

```

Comment:
You actually need to enclose this in an arrow function to ensure that you maintain its `this` context.